uswds/uswds
# USWDS - Header: Search input appears last in menu for mobile header.

# Summary
**Removed the CSS order property from mobile header navigation.** Now, the visual order of the component matches the tab order. 

## Details
This change visually moves the search bar to appear last in the list of menu items in the mobile header. This change results in the header being compliant with [WCAG criterion 2.4.3 (Focus order)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-order) in mobile view.


## Breaking change

:warning: This is a breaking change.

This change causes the search bar to appear at the bottom of the menu. If you would like to visually keep the search bar at the top of the menu, you will need to reorder your markup in the mobile view.

## Related issue

Closes #5971, closes #3526, closes #3920.



## Related pull requests

Changelog uswds/uswds-site#2798

## Preview link

Preview link: [Header Component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cb-remove-order-from-usa-nav/?path=/story/components-header--default)


## Problem statement

_Summarize the problem this PR solves in a clear and concise statement._

Web pages should be able to be navigated in an order that preserves meaning and operability. Currently, the tab order in the mobile header menu does not match the visual order. Therefore, the component does not meet [WCAG criterion 2.4.3 (Focus order)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-order).


## Solution


By removing the `order: 2; style` from `.usa-nav__primary`, the search bar appears at the bottom of the menu in the mobile header, maintaining tab order and preserving meaning. This approach was chosen based on a [conversation in dev sync on 8/6/24](https://docs.google.com/document/d/172TU5BVS9kGZQDuPcAhGoZjaTd2ngzcTayq4WRRTBUs/edit#heading=h.rcadppdhx7dt). This will impact users that wish to visually keep the search bar at the top of the menu; these users will need to reorder their markup in the mobile view.

## Testing and review


1. In Storybook, select the header component.
2. Resize the browser to minimum width so that the menu is collapsed.
3. Select the "menu" button and validate that the search input field appears after other sample links in the menu.
4. Confirm the tab order is appropriate and consistent for each of the header variants.

I am looking for confirmation that the tab order makes sense and that it is consistent with [WCAG criterion 2.4.3 (Focus order)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-order),
